### ISSUE TITLE
Should follow mode be sticky?

### ISSUE THREAD

**Issue by dpalacio** [hauke96/GeoNotes#27]
Currently follow mode is interrupted by creating a note or scrolling the map. I think it is worth considering making it sticky. That is, after creating a note or briefly scrolling the map away, it should return to follow mode after a few seconds delay

**Comment by hauke96** [comment]
I like the idea of moving back to my location after I created or edited a note. This should be pretty easy to implement.

However I'm not so sure about the scrolling thing.
First I currently don't know if OsmDroid (the map framework I use) supports this kind of behavior in any helpful way.
Furthermore I'm also not fully convinced if this is a nice behavior: If I scroll away from my location then I probably want to read or check on something. I can imagine that it's pretty disturbing if the map automatically jumps back after a while.
Even if it only jumps back after I removed my finger from the screen (so after I ended the touch) I ...
1. ... may have to hold that touch for a long time depending on what I want to do (read something on the map, look around to find something, etc.)
2. ... have to wait a second or two before the map scrolls back to my location. During that time I can also simply just tap on the location-follow-button again to do that.
And as I mentioned in #26 it' currently intentional to stop location-follow mode when scrolling on the map. In my opinion that's pretty nice because I can quickly get out of this mode before I'm too far away from a location I'm interested in. And when I want to go back to my location, clicking on the location-follow button is also quite fast.

So the first thing (scroll back after note added/edited) would probably be a very nice feature but I don't think that the second one will make it. Except you can convince me from the opposite ;)

**Comment by dpalacio** [comment]
Initially I considered it wouldn't be returning to the current location inmediately after the finger is lifted, and it would be a behavior that could be learned how to be dealt with. But I have more ideas on how to handle this potentially jarring experience.

1. A warning notification could be placed on top of the map when the view is away from the current location, indicating that the mode is on and the view would eventually return to follow. Some message like:
`In follow mode`
`Follow mode is on`
2. The location cursor would pulsate in this situation hinting the user about _something_. If the current location is outside of the view, the cursor would be situated on the side, hinting the direction of the current location.
3. When the view is scrolled away from the current location a button would be displayed, maybe on the bottom of the view, offering to disable follow mode. 
`Stop following`

I understand that it is likely to be undesired behavior for some uses, but such behavior strengthens Follow Mode considerably, when you truly want the app view to follow you. I thought about this behavior while moving in a car. It bothered me much having to reenable follow mode very often.

**Comment by hauke96** [comment]
So as far as I understand, all this is just for one little feature: Automatically moving back to the current location when I lift my finger from the screen. Or have I misunderstood this?

I'm sorry but the goal of this app is to keep UI overhead down and this really adds a lot of overhead. Especially if you consider this:

1. The icon for the follow-mode already changes depending if the mode is enabled or disabled.
2. Pulsing icons or markers on the screen edge aren't technically possible on OsmDroid (it's a very simple and basic map framework).
3. This button already exists: The same button turning the follow-mode on turns it off ;)

Also I don't really want to create a lot of settings for every little corner case. Let's first see how it feels when the map moves back after a note has been created.

**Comment by dpalacio** [comment]
No, sorry. Not all of this. Either of this. I lean for the third one.
